### PR TITLE
:recycle: Uniques for g_u, g_a, u_a tables

### PR DESCRIPTION
**Pull request by AngXuDev**
Make relationships unique for groups_users, groups_allegiances, and users_allegiances tables

# Description

Make relationships unique for groups_users, groups_allegiances, and users_allegiances tables

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: data/migrations/20190815113751_groups_users.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ exports.up = function(knex) {
 			.onDelete("CASCADE")
 			.onUpdate("CASCADE");
 		groups_users.timestamps(true, true);
+
+		groups_users.unique(["user_id, group_id"]);
```
Comment (danpatrickoneill) [inline data/migrations/20190815113751_groups_users.js]: These should be two separate strings like the one above, right?

File: data/migrations/20190825172030_users_allegiances.js [hunk 2]
```diff
@@ -19,6 +19,8 @@ exports.up = function(knex) {
 			.inTable("allegiances")
 			.onDelete("CASCADE")
 			.onUpdate("CASCADE");
+
+		users_allegiances.unique(["user_id, allegiance_id"]);
```
Comment (danpatrickoneill) [inline data/migrations/20190825172030_users_allegiances.js]: Same issue as above.

File: data/migrations/20190815113751_groups_users.js (hunk 1 above)
Comment (AngXuDev) [reply data/migrations/20190815113751_groups_users.js]: Fixed, sorry about that

File: data/migrations/20190825172030_users_allegiances.js (hunk 2 above)
Comment (AngXuDev) [reply data/migrations/20190825172030_users_allegiances.js]: fixed, sorry about that

**Pull request merged**